mergify_engine/context.py

```diff
@@ -68,6 +68,7 @@ class Installation:
     repositories: "typing.Dict[github_types.GitHubRepositoryName, Repository]" = (
         dataclasses.field(default_factory=dict)
     )
+    _cache: typing.Dict[str, typing.Any] = dataclasses.field(default_factory=dict)
```

Comment:
My original idea was more about using Redis than just an internal cache.
We have access to the event to invalidate the cache so it should be ok to cache for some time.

Reply:
Also you could used:

```python
class InstallationCacheDictT(typing.TypedDict):
    team_members: typing.Dict[typing.Tuple[github_types.GitHubLogin, str], typing.List[typing.GitHubLogin]]
```